packages/snaps-controllers/src/services/webview/WebViewMessageStream.test.ts

```diff
@@ -0,0 +1,73 @@
+import { sleep } from '@metamask/snaps-utils/test-utils';
+
+import { WebViewMessageStream } from './WebViewMessageStream';
+
+describe('WebViewMessageStream', () => {
+  let mockWebView: any;
+  let getWebView: any;
+  let stream: WebViewMessageStream;
+
+  beforeEach(() => {
+    mockWebView = {
+      addEventListener: jest.fn(),
+      removeEventListener: jest.fn(),
+      injectJavaScript: jest.fn(),
+    };
+    getWebView = jest.fn().mockResolvedValue(mockWebView);
+    stream = new WebViewMessageStream({
+      name: 'testStream',
+      target: 'targetStream',
+      getWebView,
+    });
+  });
+
+  it('initializes correctly', async () => {
+    expect(getWebView).toHaveBeenCalled();
+    await sleep(1); // wait for getWebView promise to resolve
+    expect(mockWebView.addEventListener).toHaveBeenCalledWith(
+      'message',
+      expect.any(Function),
+    );
+  });
+
+  it('handles _postMessage correctly', () => {
+    stream._postMessage({ foo: 'bar' });
```

Comment:
We should test `stream.write` and not the `_` prefixed methods as those are meant to be private.

See examples here: https://github.com/MetaMask/snaps/blob/main/packages/snaps-controllers/src/services/ProxyPostMessageStream.test.ts#L8-L60

Reply:
Added.